src/main/groovy/com/codebrig/arthur/observe/structure/filter/FunctionFilter.groovy

```diff
@@ -12,18 +12,22 @@ import com.codebrig.arthur.observe.structure.StructureFilter
  */
 class FunctionFilter extends StructureFilter<FunctionFilter, Void> {
 
-    private static final Set<String> functionTypes = new HashSet<>()
-    static {
-        functionTypes.add("def") //ruby
-        functionTypes.add("FuncDecl") //go
-        functionTypes.add("MethodDeclaration") //java
-        functionTypes.add("FunctionDeclaration") //js
-        functionTypes.add("Stmt_Function") //php
-        functionTypes.add("FunctionDef") //python
+    private final MultiFilter functionFilter
+
+    FunctionFilter() {
+        super()
+        this.functionFilter = createFunctionFilter()
```

Comment:
Please get rid of all of the `createXFilter()`. Just do the necessary work in the constructor.